x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/AuthorizationEngine.java

```diff
@@ -291,6 +292,16 @@ public ActionRequestValidationException validate(ActionRequestValidationExceptio
             }
             if (index == null) {
                 validationException = addValidationError("indexPrivileges must not be null", validationException);
+            } else {
+                for (int i = 0; i < index.length; i++) {
+                    BytesReference query = index[i].getQuery();
+                    if (query != null) {
+                        validationException = addValidationError(
+                            "may only check index privileges without any DLS query [" + query.utf8ToString() + "]",
+                            validationException
+                        );
+                    }
+                }
             }
```

Comment:
This is a fix outside of the scope for this PR. I've included it because it is minor.
The DLS verification was part of the transport action not the proper request validation.
Consequently the new "profile has privilege" check missed to also run the same verification.
I have hence moved the verification as part of the request validation, where it's harder to miss.

Reply:
Aaargs there's validation in parse as well!!
I agree we should contain all of these in a single place, which is probably the _parse_ method, as you suggest.
I'm going to leave this for another time. I've had the fair share of "side quests" in the PR.